tools/extend_oem_partition_test.go

```
@@ -0,0 +1,206 @@
+			"200K",
+		}, {
+			"InvalidSize1",
+			diskName,
+			1,
+			8,
+			"-200K",
+		}, {
+			"InvalidSize2",
+			diskName,
+			1,
+			8,
+			"200T",
+		}, {
+			"InvalidSize3",
+			diskName,
+			1,
+			8,
+			"A45M",
+		}, {
+			"InvalidSize4",
+			diskName,
+			1,
+			8,
+			"+200K",
+		}, {
+			"InvalidSize5",
+			diskName,
+			1,
+			8,
+			"",
+		}, {
+			"TooLarge",
+			diskName,
+			1,
+			8,
+			"800M",
+		}, {
+			"EmptyDiskName",
+			"",
+			1,
+			8,
+			"200K",
+		},
+	}
+
+	for _, input := range testData {
+		t.Run(input.testName, func(t *testing.T) {
+			if err := ExtendOemPartition(input.disk, input.statePartNum, input.oemPartNum, input.size); err == nil {
+				t.Fatalf("error not found in test %s", input.testName)
+			}
+		})
+	}
+
+	testData2 := []struct {
+		testName     string
+		disk         string
+		statePartNum int
+		oemPartNum   int
+		size         string
+	}{
+		{
+			"SmallerSize",
+			diskName,
+			1,
+			8,
+			"60K",
+		}, {
+			"SameSize",
+			diskName,
+			1,
+			8,
+			"100K",
+		},
+	}
+
+	for _, input := range testData2 {
+		t.Run(input.testName, func(t *testing.T) {
+			if err := ExtendOemPartition(input.disk, input.statePartNum, input.oemPartNum, input.size); err != nil {
+				t.Fatalf("error in test %s", input.testName)
+			}
+		})
+	}
+
+	if err := ExtendOemPartition(diskName, 1, 8, "200K"); err != nil {
+		t.Fatal("error when extending OEM partition")
+	}
+	if err := os.Mkdir("./mt", 0777); err != nil {
+		t.Fatal("cannot create mount point")
+	}
+	defer os.Remove("./mt")
+	mountAndCheck(diskName+"p8", "This is partition 8 OEM partition", t, 180)
+	mountAndCheck(diskName+"p1", "This is partition 1 stateful partition", t, 80)
+	mountAndCheck(diskName+"p2", "This is partition 2 middle partition", t, 80)
+}
+
+func readSize(out string) int {
+	pos := 0
+	res := -1
+	var err error
+	for out[pos] != 'K' {
+		pos++
+	}
+
+	if out[pos-3] != ' ' {
+		res, err = strconv.Atoi(out[pos-3 : pos])
+		if err != nil {
+			return -1
+		}
```

Instead of swallowing the error and returning -1, it's more idiomatic to return both a value and an error. That way the caller can know why the function call failed. For example:

func readSize(out string) (int, error) {
  ...
  if err != nil {
    return -1, err
  }
}